### PR TITLE
Dynamic sparse tests

### PR DESCRIPTION
**Pull request by lucasosouza**
- Added test for pruning in DSNNMixedHeb Model
- To add: tests for coactivation calculation in MLPHeb
- Added several notebooks with experiments

### CODE REVIEW

File: tests/frameworks/pytorch/mlp_coactivations_test.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#  See the GNU Affero Public License for more details.
+#
+#  You should have received a copy of the GNU Affero Public License
+#  along with this program.  If not, see http://www.gnu.org/licenses.
+#
+#  http://numenta.org/licenses/
+#
```
Comment (subutai) [inline tests/frameworks/pytorch/mlp_coactivations_test.py]: What's this file for?

File: tests/frameworks/pytorch/hebbian_pruning_test.py [hunk 2]
```diff
@@ -0,0 +1,322 @@
+#  Numenta Platform for Intelligent Computing (NuPIC)
+#  Copyright (C) 2019, Numenta, Inc.  Unless you have an agreement
```
Comment (subutai) [inline tests/frameworks/pytorch/hebbian_pruning_test.py]: Can you move this to `nupic.research/tests/frameworks/dynamic_sparse`?

Comment (lucasosouza) [reply tests/frameworks/pytorch/hebbian_pruning_test.py]: Done

File: tests/frameworks/pytorch/mlp_coactivations_test.py (hunk 1 above)
Comment (lucasosouza) [reply tests/frameworks/pytorch/mlp_coactivations_test.py]: That is work in progress, I just left the file there as a placeholder, but will be the next commit.

**Pull request merged**